pkg/agent/agent.go

```diff
@@ -371,3 +377,15 @@ func (k *Klocksmith) getPodsForDeletion() ([]v1.Pod, error) {
 
 	return pods, nil
 }
+
+// sleepOrDone pauses the current goroutine until the done channel receives
+// or until at least the duration d has elapsed, whichever comes first. This
+// is similar to time.Sleep(d), except it can be interrupted.
+func sleepOrDone(d time.Duration, done <-chan struct{}) {
```

Comment:
... and nevermind, I was mixing it up with the ticker issue. The 'after' will be GCd in a week, so no big deal.